Modules/Loadable/Markups/MRML/vtkSlicerDijkstraGraphGeodesicPath.cxx

```diff
@@ -0,0 +1,188 @@
+{
+  this->UseScalarWeights = true;
+  this->RecalculateAdjacency = true;
+  this->CostFunction = COST_FUNCTION_DISTANCE;
+}
+
+//------------------------------------------------------------------------------
+vtkSlicerDijkstraGraphGeodesicPath::~vtkSlicerDijkstraGraphGeodesicPath()
+= default;
+
+//------------------------------------------------------------------------------
+void vtkSlicerDijkstraGraphGeodesicPath::PrintSelf(std::ostream &os, vtkIndent indent)
+{
+  Superclass::PrintSelf(os, indent);
+  os << indent << "CostFunction: " << this->GetCostFunctionAsString(this->CostFunction) << std::endl;
+}
+
+//------------------------------------------------------------------------------
+const char* vtkSlicerDijkstraGraphGeodesicPath::GetCostFunctionAsString(int costFunction)
+{
+  switch (costFunction)
+    {
+    case COST_FUNCTION_DISTANCE:
+      {
+      return "distance";
+      }
+    case COST_FUNCTION_ADDITIVE:
+      {
+      return "additive";
+      }
+    case COST_FUNCTION_INVERSE_SQUARED:
+      {
+      return "inverseSquared";
+      }
+    default:
+      {
+      return "";
+      }
+    }
+}
+
+//------------------------------------------------------------------------------
+const char* vtkSlicerDijkstraGraphGeodesicPath::GetCostFunctionAsHumanReadableString(int costFunction)
+{
+  switch (costFunction)
+    {
+    case COST_FUNCTION_DISTANCE:
+      {
+      return "Distance";
+      }
+    case COST_FUNCTION_ADDITIVE:
+      {
+      return "Additive";
+      }
+    case COST_FUNCTION_INVERSE_SQUARED:
+      {
+      return "Inverse squared";
+      }
+    default:
+      {
+      return "";
+      }
+    }
+}
+
+//----------------------------------------------------------------------------
+int vtkSlicerDijkstraGraphGeodesicPath::RequestData(
+  vtkInformation* vtkNotUsed(request),
+  vtkInformationVector** inputVector,
+  vtkInformationVector* outputVector)
+{
+  vtkInformation* inInfo = inputVector[0]->GetInformationObject(0);
+  vtkInformation* outInfo = outputVector->GetInformationObject(0);
+
+  vtkPolyData* input = vtkPolyData::SafeDownCast(
+    inInfo->Get(vtkDataObject::DATA_OBJECT()));
+  if (!input)
+    {
+    return 0;
+    }
+
+  vtkPolyData* output = vtkPolyData::SafeDownCast(
+    outInfo->Get(vtkDataObject::DATA_OBJECT()));
+  if (!output)
+    {
+    return 0;
+    }
+
+  if (this->AdjacencyBuildTime.GetMTime() < input->GetMTime() || this->RecalculateAdjacency)
+    {
+    this->Initialize(input);
+    }
+  else
+    {
+    this->Reset();
+    }
+
+  if (this->NumberOfVertices == 0)
+    {
+    return 0;
+    }
+
+  this->ShortestPath(input, this->StartVertex, this->EndVertex);
+  this->TraceShortestPath(input, output, this->StartVertex, this->EndVertex);
+  return 1;
+}
+
+//------------------------------------------------------------------------------
+void vtkSlicerDijkstraGraphGeodesicPath::BuildAdjacency(vtkDataSet* inData)
+{
+  Superclass::BuildAdjacency(inData);
```

Comment:
add a comment how it is related to the parent class implementation (what's the difference, why)